src/features/transforms/scales/useScale.tsx

```diff
@@ -0,0 +1,168 @@
+import { useCallback, useMemo } from 'react';
+
+import { ObjectData } from '@entities/types/DataTypes';
+import { getSortField } from '../sorting/sort';
+import { SortBy } from '../sorting/SortTypes';
+import { ScaleBy } from './ScaleTypes';
+
+type Props = { objects: ObjectData[]; scaleBy?: ScaleBy };
+
+export type ScalingFunctions = {
+    scaleBy: ScaleBy | undefined;
+    getScale: (object: ObjectData) => number; // returns radius multiplier (to be multiplied by scalar)
+    maxValue: number;
+    minValue: number;
+};
+
+const useScale = ({ objects, scaleBy }: Props): ScalingFunctions => {
+    // If caller didn't pass, they'd use page params via usePageParams normally
+    const minValue = getMinimumValue(scaleBy);
+    const maxValue = useMemo(() => getMaximumValue(objects, scaleBy), [objects, scaleBy]);
+
+    const shouldUseLogScale = shouldUseLogarithmicScale(scaleBy);
+
+    const needsSqrtTransform =
+        scaleBy === SortBy.Area ||
+        scaleBy === SortBy.Population ||
+        scaleBy === SortBy.PopulationAttested ||
+        scaleBy === SortBy.PopulationOfDescendants;
+
+    const transformValue = (v: number) => (needsSqrtTransform ? Math.pow(v, 0.5) : v);
+
+    const tMin = transformValue(minValue);
+    const tMax = transformValue(maxValue);
+
+    const range = shouldUseLogScale ? Math.log10(Math.max(1, tMax - tMin)) : tMax - tMin;
+
+    const getNormalizedValue = useCallback(
+        (value: number | string): number => {
+            let numericValue: number;
+            if (typeof value === 'number') numericValue = value;
+            else numericValue = Number(value) || 0;
+
+            numericValue = transformValue(numericValue);
+
+            if (tMax === tMin) return 1;
+            if (numericValue > tMax) return 1;
+
+            numericValue -= tMin;
+            if (numericValue <= 0) return 0;
+            if (shouldUseLogScale) return Math.log10(numericValue) / range;
+            return numericValue / range;
+        },
+        [scaleBy, tMin, tMax, range, shouldUseLogScale],
+    );
+
+    const getScale = useCallback(
+        (object: ObjectData) => {
+            if (!scaleBy || scaleBy === 'None') return 2; // default radius multiplier
+
+            const val = getSortField(object, scaleBy);
+            if (val == null) return 0; // not renderable
+
+            const normalized = getNormalizedValue(val as number);
+            // Map normalized 0..1 to radius multiplier 2..10
+            return 2 + normalized * 8;
+        },
+        [scaleBy, getNormalizedValue],
+    );
+
+    return { scaleBy, getScale, maxValue, minValue };
+};
+
+export default useScale;
+
+function getMinimumValue(scaleBy?: ScaleBy): number {
```

Comment:
I'd move the functions you copied from `useColor` and just put it in the `@features/transforms` directory (the parent one to this) make a separate `getMinimumValue.ts` and `getMaximumValue.ts` (or maybe put them in the same file?)

But how do you handle both a ScaleBy and a ColorBy? very optimistically think you can do the suggestions below.
```suggestion
export function getMinimumValue(field?: ColorBy | ScaleBy): number {
```